libminifi/include/FlowController.h

```diff
@@ -149,6 +149,34 @@ class FlowController : public core::controller::ForwardingControllerServiceProvi
   std::map<std::string, std::unique_ptr<io::InputStream>> getDebugInfo() override;
 
  private:
+  class UpdateState {
+    class UpdateLock;
+   public:
+    UpdateLock getUpdateLock() { return UpdateLock(*this); }
+    bool isUpdating() const { return update_block_count_ > 0; }
+
+   private:
+    class UpdateLock {
+     public:
+      UpdateLock(UpdateState& update_state) : update_state_(update_state) {
+        update_state_.beginUpdate();
+      }
+
+      ~UpdateLock() {
+        update_state_.endUpdate();
+      }
+     private:
+      UpdateState& update_state_;
+    };
+
```

Comment:
Please check the rule of 5 for this class. I think copying the object will result in double decrease and underflowing update count.

Reply:
The latter sounds good, thanks, updated in 1fd1685fbccb60386133baeee87ddcffa4747487